RFS/src/main/java/com/rfs/cms/CmsClient.java

```diff
@@ -41,5 +42,46 @@ public interface CmsClient {
      * Updates the Metadata Migration entry in the CMS.  Returns an Optional; if the document was updated,
      * it will be the updated entry and empty otherwise.
      */
-    public Optional<CmsEntry.Metadata> updateMetadataEntry(CmsEntry.MetadataStatus status, String leaseExpiry, Integer numAttempts);
+    public Optional<CmsEntry.Metadata> updateMetadataEntry(CmsEntry.Metadata newEntry, CmsEntry.Metadata lastEntry);
```

Comment:
Lets defer making functionality choices on SonarLint guidance for the moment.  I'm often against using `@Nullable` / `@NonNull` since forgetting the annotation is trivial and can cause runtime failures.